src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -554,7 +554,8 @@ def do_compile(self,
       raise TaskError("Compilation failure: {}".format(e))
 
   def _record_compile_classpath(self, classpath, targets, outdir):
-    text = '\n'.join(classpath)
+    relative_classpaths = [os.path.relpath(path) for path in classpath]
```

Comment:
This is implicitly relativizing to `cwd`: should relativize to the workdir probably.

But also: should use `fast_relpath`, which 1) is a bit faster, 2) enforces that what you're relativizing is a subdir of the thing you're relativizing to.